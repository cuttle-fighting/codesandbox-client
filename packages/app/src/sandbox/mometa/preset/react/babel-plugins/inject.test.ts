import { transform } from '@babel/core';
import babelPluginMometaReactInject from './inject';

describe('babel-inject', () => {
  it('spec', function () {
    const rs = transform(
      `
    const elem = <div>
      <h1 title={'abc'}>Hello World</h1>
      <Tabs>
      <Tabs.TabPane key={'tool'} tab={'物料'}>
        <p>物料</p>
        <p className='empty'></p>
      </Tabs.TabPane>
      <Tabs.TabPane key={'attr'} tab={'属性'}></Tabs.TabPane>
      </Tabs>
    </div>
    `,
      {
        filename: '/file.jsx',
        parserOpts: {
          plugins: ['jsx'],
        },
        babelrc: false,
        plugins: [babelPluginMometaReactInject],
      }
    );

    expect(rs.code).toMatchSnapshot();
  });
});
