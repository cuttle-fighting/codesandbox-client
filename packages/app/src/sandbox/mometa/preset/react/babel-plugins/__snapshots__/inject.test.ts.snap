// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-inject spec 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

var _emptyChildrenPlaceholder = _interopRequireDefault(require(\\"@@__moment/empty-children-placeholder\\"));

var elem = /*#__PURE__*/React.createElement(\\"div\\", {
  __mometa: {
    \\"start\\": {
      \\"line\\": 2,
      \\"column\\": 17
    },
    \\"end\\": {
      \\"line\\": 11,
      \\"column\\": 10
    },
    \\"name\\": \\"div\\",
    \\"text\\": \\"<div>\\\\n      <h1 title={'abc'}>Hello World</h1>\\\\n      <Tabs>\\\\n      <Tabs.TabPane key={'tool'} tab={'物料'}>\\\\n        <p>物料</p>\\\\n        <p className='empty'></p>\\\\n      </Tabs.TabPane>\\\\n      <Tabs.TabPane key={'attr'} tab={'属性'}></Tabs.TabPane>\\\\n      </Tabs>\\\\n    </div>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, /*#__PURE__*/React.createElement(\\"h1\\", {
  title: 'abc',
  __mometa: {
    \\"start\\": {
      \\"line\\": 3,
      \\"column\\": 6
    },
    \\"end\\": {
      \\"line\\": 3,
      \\"column\\": 40
    },
    \\"name\\": \\"h1\\",
    \\"text\\": \\"<h1 title={'abc'}>Hello World</h1>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, \\"Hello World\\"), /*#__PURE__*/React.createElement(Tabs, {
  __mometa: {
    \\"start\\": {
      \\"line\\": 4,
      \\"column\\": 6
    },
    \\"end\\": {
      \\"line\\": 10,
      \\"column\\": 13
    },
    \\"name\\": \\"Tabs\\",
    \\"text\\": \\"<Tabs>\\\\n      <Tabs.TabPane key={'tool'} tab={'物料'}>\\\\n        <p>物料</p>\\\\n        <p className='empty'></p>\\\\n      </Tabs.TabPane>\\\\n      <Tabs.TabPane key={'attr'} tab={'属性'}></Tabs.TabPane>\\\\n      </Tabs>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, /*#__PURE__*/React.createElement(Tabs.TabPane, {
  key: 'tool',
  tab: '物料',
  __mometa: {
    \\"start\\": {
      \\"line\\": 5,
      \\"column\\": 6
    },
    \\"end\\": {
      \\"line\\": 8,
      \\"column\\": 21
    },
    \\"name\\": \\"Tabs.TabPane\\",
    \\"text\\": \\"<Tabs.TabPane key={'tool'} tab={'物料'}>\\\\n        <p>物料</p>\\\\n        <p className='empty'></p>\\\\n      </Tabs.TabPane>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, /*#__PURE__*/React.createElement(\\"p\\", {
  __mometa: {
    \\"start\\": {
      \\"line\\": 6,
      \\"column\\": 8
    },
    \\"end\\": {
      \\"line\\": 6,
      \\"column\\": 17
    },
    \\"name\\": \\"p\\",
    \\"text\\": \\"<p>物料</p>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, \\"\\\\u7269\\\\u6599\\"), /*#__PURE__*/React.createElement(\\"p\\", {
  className: \\"empty\\",
  __mometa: {
    \\"start\\": {
      \\"line\\": 7,
      \\"column\\": 8
    },
    \\"end\\": {
      \\"line\\": 7,
      \\"column\\": 33
    },
    \\"name\\": \\"p\\",
    \\"text\\": \\"<p className='empty'></p>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, /*#__PURE__*/React.createElement(_emptyChildrenPlaceholder[\\"default\\"], {
  __mometa: {
    \\"start\\": {
      \\"line\\": 7,
      \\"column\\": 8
    },
    \\"end\\": {
      \\"line\\": 7,
      \\"column\\": 33
    },
    \\"name\\": \\"p\\",
    \\"text\\": \\"<p className='empty'></p>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}))), /*#__PURE__*/React.createElement(Tabs.TabPane, {
  key: 'attr',
  tab: '属性',
  __mometa: {
    \\"start\\": {
      \\"line\\": 9,
      \\"column\\": 6
    },
    \\"end\\": {
      \\"line\\": 9,
      \\"column\\": 59
    },
    \\"name\\": \\"Tabs.TabPane\\",
    \\"text\\": \\"<Tabs.TabPane key={'attr'} tab={'属性'}></Tabs.TabPane>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}, /*#__PURE__*/React.createElement(_emptyChildrenPlaceholder[\\"default\\"], {
  __mometa: {
    \\"start\\": {
      \\"line\\": 9,
      \\"column\\": 6
    },
    \\"end\\": {
      \\"line\\": 9,
      \\"column\\": 59
    },
    \\"name\\": \\"Tabs.TabPane\\",
    \\"text\\": \\"<Tabs.TabPane key={'attr'} tab={'属性'}></Tabs.TabPane>\\",
    \\"filename\\": \\"/file.jsx\\"
  }
}))));"
`;
